<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" dir="ltr" lang="en">
<head>
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
	<meta name="keywords" content="updating CDF fields using dal.set.field, baan,baanerp,erp,forum,discussion,bulletin board" />
	<meta name="description" content="[Archive] updating CDF fields using dal.set.field Tools Development" />
	
	<title>updating CDF fields using dal.set.field [Archive]  - Baanboard.com</title>
	<link rel="stylesheet" type="text/css" href="../styles.css" />
</head>
<body>
<div class="pagebody">
<div id="navbar"><a href="../index.html">Newbaanboard</a> &gt; <a href="../index.html">Baan Quick Support: Functional &amp; Technical</a> &gt; <a href="../index.html">Tools Development</a> &gt; updating CDF fields using dal.set.field</div>
<hr />
<div class="pda"></div>

<hr />

<div class="post"><div class="posttop"><div class="username">IndoTech</div><div class="date">27th June 2016, 13:22</div></div><div class="posttext">Hi <br />
<br />
can't  we update CDF fields like below?<br />
 dal.set.field(&quot;tdsls401.cdf_pcd1&quot;, vlsls401.pcd1)<br />
<br />
 am getting error or ret value for  ret = dal.save.object(&quot;tdsls401&quot;)<br />
= -12. what it means?<br />
<br />
code :<br />
ret1  = dal.get.object(&quot;tdsls401&quot;,TRUE,&quot;tdsls401.orno&quot;,vlsls401.orno, &quot;tdsls401.pono&quot;,vlsls401.pono)<br />
<br />
		dal.change.object(&quot;tdsls401&quot;) <br />
		dal.set.field(&quot;tdsls401.cdf_pcd1&quot;, vlsls401.pcd1)<br />
		dal.set.field(&quot;tdsls401.cdf_pcd2&quot;, vlsls401.pcd2)<br />
<br />
		ret = dal.save.object(&quot;tdsls401&quot;)<br />
<br />
Regards<br />
Keadr</div></div><hr />


<div class="post"><div class="posttop"><div class="username">bhushanchanda</div><div class="date">27th June 2016, 17:56</div></div><div class="posttext">Hi,<br />
<br />
What error message do you get with - <br />
<br />
string err(1000)<br />
dal.get.error.message(err)<br />
<br />
What kind of CDF is it? Is the value according to the data type?</div></div><hr />


<div class="post"><div class="posttop"><div class="username">DDPatel</div><div class="date">20th April 2021, 15:32</div></div><div class="posttext">i am getting same error...<br />
<br />
Has any one know about this error?<br />
<br />
	long ret<br />
	string error.msg(200)<br />
	db.retry.point()<br />
	select tdsls401.*<br />
	from tdsls401 for update<br />
	where tdsls401._index1 = {:tdcsl956.orno, :tdcsl956.pono, :tdcsl956.sqnb}<br />
	selectdo	<br />
		dal.change.object(&quot;tdsls401&quot;) <br />
			dal.set.field(&quot;tdsls401.cdf_absl&quot;,tcyesno.yes) |absl is possible output Yes/No<br />
			ret = dal.save.object(&quot;tdsls401&quot;) |ret = -12<br />
			if ret = 0 then <br />
				commit.transaction()<br />
			else<br />
				abort.transaction()<br />
				dal.get.error.message(error.msg) |&quot;Register number sequecne not exist for XXX Business&gt;&gt;&gt;&gt;&quot;[[but actually its maintained already]]<br />
			endif<br />
		|endif		<br />
	endselect<br />
<br />
<br />
I don't know where i am making mistake...<br />
<br />
Regards,<br />
DDPatel</div></div><hr />



</div>
</body>
</html>